### PR TITLE
Simple Mode Collector

### PR DESCRIPTION
**Pull request by TheConnMan**
This simple mode collector collects the following metrics:
- netuitive.linux.cpu.total.utilization.percent
- netuitive.linux.memory.utilizationpercent
- netuitive.linux.diskspace.avg_byte_percentused
- netuitive.linux.iostat.max_util_percentage

If anyone knows how to run the tests for this project please let me know.

### CODE REVIEW

**Review by bspindler — APPROVED** [review]
looks good!

File: src/collectors/simple/simple.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        # Derivatives take one cycle to warm up
+        if total != 0:
+            self.publish('cpu.total.utilization.percent', (total - idle) / total * 100)
+
+    def collect_memory_proc(self, lines):
+        # Compute and convert all memory usage in bytes
+        total = self.memory_proc_line_to_bytes(lines[0])
+        free = self.memory_proc_line_to_bytes(lines[1])
+        buffers = self.memory_proc_line_to_bytes(lines[3])
+        cached = self.memory_proc_line_to_bytes(lines[4])
```
Comment (andrewpaine) [inline src/collectors/simple/simple.py:133]: Are you sure you can rely on the line sequence to get these values? 
e.g., according to this https://www.centos.org/docs/5/html/5.1/Deployment_Guide/s2-proc-meminfo.html centos is different.

File: src/collectors/simple/simple.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        softirq = self.derivative('cpu.total.softirq', long(elements[7]), diamond.collector.MAX_COUNTER)
+        steal = self.derivative('cpu.total.steal', long(elements[8]), diamond.collector.MAX_COUNTER)
+        guest = self.derivative('cpu.total.guest', long(elements[9]), diamond.collector.MAX_COUNTER)
+        guest_nice = self.derivative('cpu.total.guest_nice', long(elements[10]), diamond.collector.MAX_COUNTER)
+
+        total = sum([user, nice, system, idle, iowait, irq, softirq, steal, guest, guest_nice])
+
+        # Derivatives take one cycle to warm up
+        if total != 0:
+            self.publish('cpu.total.utilization.percent', (total - idle) / total * 100)
```
Comment (andrewpaine) [inline src/collectors/simple/simple.py:113]: Do we have to keep the full tree for this metric? 
It seems odd to have 3 levels for this metric but only 1 for the others.

Comment (TheConnMan) [reply src/collectors/simple/simple.py:113]: This is to replicate https://github.com/netuitive-community-packages/netuitive-packages-linux/blob/4393a315e6261284fd3985c182ccec9be6694cb5/analyticConfigurations/computed_metric-linux.json#L24, so yes.

File: src/collectors/simple/simple.py (hunk 1 above)
Comment (TheConnMan) [reply src/collectors/simple/simple.py:133]: Good point. I'll map all the lines into a dict then pull them out by name.

**Comment by TheConnMan** [comment]
I've narrowed down why the Disk I/O stat metrics aren't matching up. I won't be able to get them to match exactly with a single metric incoming. The computed metric (https://github.com/netuitive-community-packages/netuitive-packages-linux/blob/4393a315e6261284fd3985c182ccec9be6694cb5/analyticConfigurations/computed_metric-linux.json#L138-L145) takes the max of the 5 min rollups for disk util percent. This is a max of averages. My simple mode metric value is a max across disks, then Metricly averages the values in a 5 min rollup. An average of maxes is different than a max of averages. I verified the raw numbers are correct by looking at the max of maxes as well as local computations in a test agent.

Example:
- `max([avg([5, 0, 0, 0, 0]), avg([10, 0, 0, 0, 0])]) = 2`
- `avg([max([5, 0, 0, 0, 0]), max([10, 0, 0, 0, 0])]) = 7.5`

This difference is most pronounced when the raw data has spikes like the example above.

This being said, I think having this difference is alright for simple mode. It still provides value and will become more accurate once simple mode is disabled.

**Pull request merged**